HIP/hip-657.md

```diff
@@ -0,0 +1,115 @@
+---
+
+
+## Abstract
+
+
+Currently, there is no native way to update the metadata of a non-fungible token (NFT) on the Hedera network. This HIP proposes the creation of a new key called the METADATA key, which will allow updating NFT metadata. This will provide the Hedera network with functional parity with other networks and enable developers to create dynamic NFTs with new interactions.
+
+
+## Motivation
+
+
+Dynamic NFTs are a valuable addition to the world of non-fungible tokens because they allow for more interactive and immersive experiences. With the ability to update the metadata of an NFT, developers can create game pieces that gain new equipment and stats, legal agreements that reflect changing parameters, or even NFTs that update their images based on the time of day. Dynamic NFTs also open up the possibility for NFTs to be used in a wider range of industries and applications, such as supply chain tracking, real estate, among others. The potential for dynamic NFTs is numerous, and their introduction will greatly enhance the capabilities of the NFT ecosystem.
+
+
+## Rationale
+
+
+To address this issue, the proposed solution is to create a new key type called the METADATA key. This key will allow NFTs in a collection to update their metadata.
+
+
+The use of the METADATA key to update NFT metadata will not affect the functionality of existing NFTs on the Hedera Network. It will be available for new and existing NFTs.
+
+
+## User stories
+
+
+- As a collector of NFTs, I want to be able to update the metadata of an NFT in my collection so that I can add new information or make changes to the existing metadata.
+
+
+- As a creator of NFTs, I want to be able to update the metadata of an NFT while leaving an auditable on-chain history of when the update happened and which key was used to sign for the update action so that I can retain as much transparency as possible. 
+
+
+- As a creator of NFTs, I want to use IPFS to manage my dynamic NFT’s metadata so that I don’t have to use a centralized file storage solution to enable the ability to edit.  
+
+
+- As a developer building applications on top of the Hedera Network, I want the ability to update the metadata of NFTs so that I can create dynamic NFTs and build more advanced use cases.
+
+
+## Specification
+
+
+The METADATA key will be a new key that allows for updating the metadata of an NFT in a new NftMetadataUpdate transaction.
+
+The NftMetadataUpdate transaction should accept the following parameters:
+	- identifier (string): the 0.0.<token-type>.<serial-no>
+- the new NFT metadata: the new metadata for the NFT
+
+
+The NftMetadataUpdate transaction should have a new token type field called "dynamic_metadata" of type bool, which is valid only for non-fungible tokens. By default, this field is set to false (meaning the metadata for this token type is immutable). If set to true, owners of this token type can change their NFT's metadata via the NftMetadataUpdate transaction.
```

Comment:
To confirm does this imply only the current owner can make the change.
If so might be good to call this out under Security section.
Explicitly note the permissions consideration that token creators and previous owners cannot make such changes

Reply:
as with the other keys on a token_id, the account which controls they associated pub/priv key pair is able to sign for the `NftMetadataUpdate` action. 

Unless i've misunderstood something, there's no requirement to have both custody of the token_id (or nft_id) && control the pub/priv key pair in order to make a successful `NftMetadataUpdate`. 

Since the METADATA key is defined on the token_id, rather than on the nft_id, individual NFT holders will not be able to update each NFT which they custody unless of course they also maintain the `METADATA` private key.

Reply:
@Ashe-Oro is right. The intention here was that only the entity in control of this key would be able to call `NftMetadataUpdate`. Individual holders would be unable to update their metadata unless they knew that key.